src/framework/components/element/text-element.js

```diff
@@ -222,34 +229,120 @@ Object.assign(pc, function () {
         _updateText: function (text) {
             var i;
             var len;
+            var results;
+            var tags;
 
             if (text === undefined) text = this._text;
 
-            var symbols = pc.string.getSymbols(text);
+            // get the list of symbols
+            this._symbols = pc.string.getSymbols(text);
+
+            // handle null string
+            if (this._symbols.length === 0) {
+                this._symbols = [" "];
+            }
+
+            // extract markup
+            if (this._enableMarkup) {
+                results = pc.evaluateMarkup(this._symbols);
+                this._symbols = results.symbols;
+                tags = results.tags;
+            }
 
-            if (this.rtlReorder) {
+            // handle LTR vs RTL ordering
+            if (this._rtlReorder) {
                 var rtlReorderFunc = this._system.app.systems.element.getRtlReorder();
                 if (rtlReorderFunc) {
-                    symbols = rtlReorderFunc(symbols);
+                    results = rtlReorderFunc(this._symbols);
+
+                    this._rtl = results.rtl;
+
+                    // reorder symbols according to unicode reorder mapping
+                    this._symbols = results.mapping.map(function (v) {
+                        return this._symbols[v];
+                    }, this);
+
+                    // reorder tags if they exist, according to unicode reorder mapping
+                    if (tags) {
+                        tags = results.mapping.map(function (v) {
+                            return tags[v];
+                        });
+                    }
                 } else {
                     console.warn('Element created with rtlReorder option but no rtlReorder function registered');
                 }
-            }
+            } else {
+                this._rtl = false;
+            }
+
+            // resolve color tags
+            if (tags) {
+                // color dictionary - temp
+                var dict = {
```

Comment:
Is this supposed to be in the PR or should we comment it out until there is a proper palette implemented?

Reply:
ok removed